### DISCUSSION TITLE
Is it a good extension to enable named zustand store?

### DISCUSSION THREAD

**Discussion by ZhengRui** [pmndrs/zustand#1148] [Ideas]
## Why named store

Zustand store is already awesome, it's very clean and simple to use. By default, the store context is global for the whole app, so each store only has one instance and all components connected to it are synced. To scope the syncing, so components within a group are synced and independant among different groups, zustand also provides [Context+Provider](https://github.com/pmndrs/zustand#react-context). However, using `<Provider></Provider>` to scope a group of synced components constrains the layout design, it can not handle cases when layout is more dynamic thus cannot be predetermined.

Why not using Jotai or Recoil? Their granularity can be too thin, and many like me might prefer zustand APIs, and for legacy projects using external stores, it is easier to convert to zustand. A named zustand store strikes a good balance between the granularity and ease of use.

I had a POC implementation of named store at [here](https://github.com/ZitySpace/named-zustand-store), but I wanna know more thoughts on this idea, maybe I totally miss some limitations or better practice of achieving this.

**Comment by dai-shi** [comment]
So, it's a stores-in-store pattern. This is possible thanks to the new `createStore` and `useStore` apis.
